### ISSUE TITLE
[BUG] Hot Reload not working .net 8 for .razor files

### ISSUE THREAD

**Issue by mprattinger** [microsoft/vscode-dotnettools#724]
### Describe the Issue

I am using and .net blazor server app updated to .net 8 on mac in vscode. The hot reload symbol is showing in the debug bar, in the hot reload console it writes:

💡 Possible changes were found.

but when i save and press the hot reload button nothing happens.

I saw the same behavior on windows and on mac.

r Michael

### Steps To Reproduce

.net 8 mac -> dotnet new blazorserver
vs code with c# dev kit
F5 to start debug
change something on Index.razor

### Expected Behavior

The changes made in the page should show in the browser

### Environment Information

macOS
vs code 1.84.2
c# dev kit v1.0.14

**Comment by WardenGnaw** [comment]
This is a known limitation, our docs are currently updating but it should appear at https://code.visualstudio.com/docs/csharp/debugging#hot-reload

![image](https://github.com/microsoft/vscode-dotnettools/assets/3953714/78322c32-86b1-4a32-b426-6c8a26cd1924)

**Comment by mprattinger** [comment]
oh,

thanks for the answer, havn't seen this before

hopefully this will be availiable soon

**Comment by dibarbet** [comment]
I think this needs to go to Razor to take a look.

**Comment by JohnGalt1717** [comment]
This still is an issue and makes the dev inner loop almost impossible to tollerate.

**Comment by schaveyt** [comment]
Solution:

1. Open the terminal (`Ctr ~`
2. Type `dotnet watch`

In my experience, this is how the majority of C# team who use Visual Studio Code operate. For those who what the Visual Studio experience, usual Visual Studio. Get the UI out of the way, so the engineer has more control.

**Comment by ryzngard** [comment]
This is a known issue. @phil-allen-msft  do we have a tracking issue for this that is on GH?

**Comment by ktravelet** [comment]
getting this while using .NET MAUI v0.12.42 (pre-release). Does this not work for Mac? 

"Changes made in project 'MauiApp1 (net8.0-maccatalyst)' require restarting the application: Changes are not allowed with the targeted ARM64 CLR version. Consider switching to .NET 7 or newer."

**Comment by ryzngard** [comment]
Blocked until https://github.com/dotnet/razor/issues/10341 is done

@ktravelet please file a separate issue for `.NET MAUI` not working

**Comment by RezaRahmatzadeh** [comment]
When can we expect hot reload functionality for .cshtml files in ASP.NET MVC Core projects using the C# Dev Kit in VS Code?

**Comment by webreidi** [comment]
@RezaRahmatzadeh we have been working to make Hot Reload work better and get it out of Experimental mode in C# Dev Kit for quite some time. The latest is the rollout of https://github.com/dotnet/razor/issues/10341, which completed recently. We'd love to hear if your experience with hot reload with .cshtml files has improved.

**Comment by jtlowe** [comment]
@webreidi what is the best way to test if this? My asp.net core .cshtml views are still not being updated, even when Hot Reload says `Code changes were successfully applied.` 

I'm using the latest VS Code Insider version on Mac with the pre-release C# Dev Kit For Visual Studio Code (last updated 2025-03-18). No other extensions installed.

Thank you for working to improve this! I'm really looking forward to having this feature.

**Comment by ryzngard** [comment]
It looks like MVC changes aren't being reflected still. I'm assuming that's what you're using @jtlowe ? I am seeing blazor hot reload working except in WASM experiences

**Comment by jtlowe** [comment]
@ryzngard you’re correct. My ASP.NET Core 9 MVC razor view .cshtml file changes are still not being reflected when I reload the page. 

I’m hopeful now that Blazor’s reload is working! I currently have to run `dotnet watch run` separately and attach to a process to run. I’m getting used to it, but it’s not as simple as what I’m used to with Visual Studio.

**Comment by ivanjx** [comment]
also not working with blazor wasm standalone projects. the changes in .razor files are recognized but nothing happens on the actual app page itself. works fine if i use `dotnet watch` but then i dont have debugging.

**Comment by claudiaregio** [comment]
@mprattinger Are you still seeing this issue?

**Comment by RezaRahmatzadeh** [comment]
> [@mprattinger](https://github.com/mprattinger) [@WardenGnaw](https://github.com/WardenGnaw) [@isadorasophia](https://github.com/isadorasophia) [@dibarbet](https://github.com/dibarbet) [@JohnGalt1717](https://github.com/JohnGalt1717) [@schaveyt](https://github.com/schaveyt) [@ryzngard](https://github.com/ryzngard) [@ktravelet](https://github.com/ktravelet) [@RezaRahmatzadeh](https://github.com/RezaRahmatzadeh) [@webreidi](https://github.com/webreidi) [@jtlowe](https://github.com/jtlowe) [@ivanjx](https://github.com/ivanjx) [@claudiaregio](https://github.com/claudiaregio) [@artyomszasa](https://github.com/artyomszasa) [@Eptagone](https://github.com/Eptagone)
> 
> A fix for this issue is now available in the next pre-releases of C#DK/C#. If you'd like to try out the fix, please see [these instructions](https://github.com/microsoft/vscode-dotnettools/wiki/Installing-and-updating-pre%E2%80%90release-versions-of-C%23-Dev-Kit-and-C%23-Extension) to update or try pre-release versions.

I tried running it, but the web app fails to start on localhost due to a permission error. It was running fine before with the same code and configurations. Could this be related to the latest Windows update? If so, why does it still run fine in Visual Studio?

**Comment by RezaRahmatzadeh** [comment]
Voilà! I tried it again after updating Windows; the localhost port issue is gone. Hot reload in VS Code (with the latest extensions) works like a charm in my ASP.NET MVC .NET 9 project. The UI updates instantly after saving changes. Thank you, Microsoft team!

**Comment by RezaRahmatzadeh** [comment]
I've been using this recently and overall everything is working well, including hot reload. The only issue I've noticed is with VS Code constantly using around 10% CPU even when the app is stopped and the terminal is killed & closed. My Dell laptop fan keeps running non-stop, and the CPU usage isn’t released from VS Code. It might be related to the C# DevKit, hot reload, or something similar, just thought it was worth mentioning based on my testing.

**Comment by LittleLittleCloud** [comment]
@RezaRahmatzadeh Could you enable Diagnostics and share the log from .NET HotReload channel, wondering if it relevant to file watcher issue we are recently investigating

<img width="1974" height="1212" alt="Image" src="https://github.com/user-attachments/assets/a407991c-ece2-45ea-bb3e-f2fe0fe654df" />

**Comment by RezaRahmatzadeh** [comment]
> [@RezaRahmatzadeh](https://github.com/RezaRahmatzadeh) Could you enable Diagnostics and share the log from .NET HotReload channel, wondering if it relevant to file watcher issue we are recently investigating
> 
> <img alt="Image" width="1974" height="1212" src="https://private-user-images.githubusercontent.com/16876986/513591060-a407991c-ece2-45ea-bb3e-f2fe0fe654df.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMwOTI5MDksIm5iZiI6MTc2MzA5MjYwOSwicGF0aCI6Ii8xNjg3Njk4Ni81MTM1OTEwNjAtYTQwNzk5MWMtZWNlMi00NWVhLWJiM2UtZjJmZTBmZTY1NGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE0VDAzNTY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5MWEwYWE0Y2E3NWI5NmI4ODlkOTM2YzU2ZTM2NDRjNzU4ZmJjMjQ1NDllYTBiNDlkYTJlNzJiYjYwNGRmZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sadaViuYfRWOfAY3ijmKkjSEA-whGnNJbgT1DemwOxE">

I tried using that log level for the hot reload channel. There were no exceptions, and everything looks fine. I’ll keep monitoring it throughout my work, and if anything shows up in the logs, I’ll share it with you.
